nowcasting_dataloader/utils/position_encoding.py

```diff
@@ -0,0 +1,291 @@
+    """
+    Normalize the geospatial coordinates by the max extant to keep everything between -1 and 1, in sin and cos
+
+    This should work on a batch level, as the geospatial bounds should be the same for every example in the batch
+
+    Args:
+        geospatial_coordinates: The coordinates for the pixels in the image
+        geospatial_bounds: The maximum extant
+
+    Returns:
+        The normalized geospatial coordinates, rescaled to between -1 and 1 for the whole extant of the training area
+
+    """
+    # Normalize the X first
+    geospatial_coordinates[0] = (geospatial_coordinates[0] - geospatial_bounds["x_min"]) / (
+        geospatial_bounds["x_max"] - geospatial_bounds["x_min"]
+    )
+    # Normalize the Y second
+    geospatial_coordinates[1] = (geospatial_coordinates[1] - geospatial_bounds["y_min"]) / (
+        geospatial_bounds["y_max"] - geospatial_bounds["y_min"]
+    )
+
+    # Now those are between 0 and 1, want between -1 and 1
+    geospatial_coordinates[0] = geospatial_coordinates[0] * 2 - 1
+    geospatial_coordinates[1] = geospatial_coordinates[1] * 2 - 1
+    # Now create a grid of the coordinates
+    # Have to do it for each individual example in the batch, and zip together x and y for it
+    to_concat = []
+    for idx in range(len(geospatial_coordinates[0])):
+        x = geospatial_coordinates[0][idx]
+        y = geospatial_coordinates[1][idx]
+        grid = torch.meshgrid(x, y)
+        pos = torch.stack(grid, dim=-1)
+        encoded_position = fourier_encode(pos, **kwargs)
+        encoded_position = einops.rearrange(encoded_position, "... n d -> ... (n d)")
+        to_concat.append(encoded_position)
+
+    # And now convert to Fourier features, based off the absolute positions of the coordinates
+    encoded_position = torch.stack(to_concat, dim=0)
+    return encoded_position
+
+
+def create_datetime_features(
+    datetimes: List[List[datetime.datetime]],
+) -> List[torch.Tensor]:
+    """
+    Converts a list of datetimes to day of year, hour of day sin and cos representation
+
+    Args:
+        datetimes: List of list of datetimes for the examples in a batch
+
+    Returns:
+        Tuple of torch Tensors containing the hour of day sin,cos, and day of year sin,cos
+    """
+    hour_of_day = []
+    day_of_year = []
+    for batch_idx in range(len(datetimes)):
+        hours = []
+        days = []
+        for index in datetimes[batch_idx]:
+            hours.append((index.hour + (index.minute / 60) / 24))
+            days.append((index.timetuple().tm_yday / 365))
+        hour_of_day.append(hours)
+        day_of_year.append(days)
+
+    outputs = []
+    for index in [hour_of_day, day_of_year]:
+        index = torch.as_tensor(index)
+        radians = index * 2 * np.pi
+        index_sin = torch.sin(radians)
+        index_cos = torch.cos(radians)
+        outputs.append(index_sin)
+        outputs.append(index_cos)
+
+    return outputs
+
+
+def encode_fouier_position(
```

Comment:
Tiny typo :slightly_smiling_face: 

Also, is this function called anywhere?

```suggestion
def encode_fourier_position(
```

Reply:
No! It was for the relative encoding, but not now, removed it